Makefile

```diff
@@ -0,0 +1,32 @@
+.PHONY: build docker test docker-test install-deps dest_dir
```

Comment:
Could we please not introduce Makefile complexity into the system again... it tends to encourage building overly complex build systems.

Reply:
Moved to single docker file, but to run tests locally require 
```docker build . --target test -t nsmgr-test-container` ```

To tag container with spire-server/spire-agent and spire.sh/spire-proxy to run tests against it, it become much faster in compare to run in docker.

Reply:
Why not just opportunistically run spire inside a test target in the single container... it removes the need for the proxy, second container complexity etc.